src/governance/object.cpp

```diff
@@ -626,7 +626,17 @@ int CGovernanceObject::CountMatchingVotes(vote_signal_enum_t eVoteSignalIn, vote
         const vote_rec_t& recVote = votepair.second;
         auto it2 = recVote.mapInstances.find(eVoteSignalIn);
         if (it2 != recVote.mapInstances.end() && it2->second.eOutcome == eVoteOutcomeIn) {
-            ++nCount;
+            Coin coin;
+            int voteWeight = 1;
+            if (!GetUTXOCoin(votepair.first, coin)) {
+                // 4x times weight vote for HPMN owners.
+                // No need to check if v19 is active since no HPMN are allowed to register before v19
+                CAmount HPMNCollateralAmount = 4000 * COIN;
+                if (coin.out.nValue == HPMNCollateralAmount) {
+                    voteWeight = 4;
+                }
+            }
+            nCount += voteWeight;
         }
```

Comment:
@nmarley Well Governance is changed here. HPMN have a 4x weighted vote.

Reply:
@ogabrielides You're right! I must have been looking at a single commit instead of the entire thing, or I didn't see the scroll window for the rest of the files.

So any votes for proposals get weighted 4x then? Yes, that will definitely have an effect on Sentinel. Thanks for catching this.

We will need a way to identify HPMNs (via RPC) and 4x those votes, _or_ (my preference) have `getcurrentvotes` RPC call return another column for "vote weight", and return 1 for regular and 4 for HPMNs.

Either way I can make the required changes in Sentinel, but would prefer that getcurrentvotes return a weight because that puts all the weight logic into Core.